File: frontend/src/types/inventory.ts
export interface CreateAssetRequest {
  asset_type: string
  name: string
  description?: string
  purchase_date: string
  purchase_price: number
  location?: string
}

export interface AssetItem {
  id: string
  name: string
  current_value: number
}

export interface CreateInventoryItemRequest {
  item_name: string
  quantity: number
  unit: string
  reorder_level: number
}

export interface InventoryItem {
  id: string
  item_name: string
  quantity: number
}

File: backend/app/infrastructure/models/inventory.py
from beanie import Document
from pydantic import Field
from typing import Optional
from datetime import datetime

class Asset(Document):
    tenant_id: str
    asset_type: str
    name: str
    description: Optional[str] = None

    purchase_date: datetime
    purchase_price: float
    current_value: float

    depreciation_rate: float = 0.1

    assigned_to: Optional[str] = None
    location: Optional[str] = None

    class Settings:
        name = "assets"

class Inventory(Document):
    tenant_id: str
    item_name: str
    quantity: int
    unit: str

    reorder_level: int

    last_updated: datetime = Field(default_factory=datetime.utcnow)

    class Settings:
        name = "inventory"

class MaintenanceSchedule(Document):
    tenant_id: str
    asset_id: str

    maintenance_date: datetime
    maintenance_type: str

    performed_by: Optional[str] = None
    status: str = "scheduled"

    class Settings:
        name = "maintenance_schedules"
